jsonargparse_tests/test_typehints.py

```diff
@@ -489,6 +492,42 @@ def test_mapping_nested_without_args(parser):
     assert {"b": {"c": 2}} == parser.parse_args(['--map={"b": {"c": 2}}']).map
 
 
+def test_typeddict_without_arg(parser):
+    if sys.version_info < (3, 8):
+        pytest.skip("TypedDict introduced in python 3.8")
+    parser.add_argument("--typeddict", type=TypedDict("MyDict", {}))
+    assert {} == parser.parse_args(["--typeddict={}"])["typeddict"]
+    pytest.raises(ArgumentError, lambda: parser.parse_args(['--typeddict={"a":1}']))
```

Comment:
Better if you assert part of the exception messages. For example in this case, it could be asserted that the error says `Unexpected keys`. For how to do it, see for example https://github.com/omni-us/jsonargparse/blob/af37f088f257a184f1f9e77812539d8a2e584276/jsonargparse_tests/test_typehints.py#L54-L56